api/all/src/main/java/io/opentelemetry/api/internal/ValidationUtil.java

```diff
@@ -18,12 +19,50 @@
 @Immutable
 public final class ValidationUtil {
 
-  private ValidationUtil() {}
+  public static final String API_USAGE_LOGGER_NAME = "io.opentelemetry.ApiUsageLogging";
```

Comment:
Seemed appropriate to log to the same logger as other API usage violations. However, instrument name violations are logged at level warning instead of at finest. I've also omitted the stack trace for now, though I could be convinced to include it.

Reply:
I'd add stack trace - may as well make it even easier to find the registration within potential string mangling through variables.

So can probably add a method to `ValidationUtil` that accepts level